apps/settings/js/panels/root/addons_item.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * This module is used to show/hide addon menuItem based on the number of
+ * current installed addons.
+ *
+ * @module AddonsItem
+ */
+define(function(require) {
+  'use strict';
+
+  var AddonsManager = require('panels/addons/addons');
+
+  function AddonsItem(element) {
+    this._enabled = false;
+    this._element = element;
+    this.addonsManager = new AddonsManager();
+    this.init();
+  }
+
+  AddonsItem.prototype = {
+    /**
+     * Set current status of addonsItem
+     *
+     * @access public
+     * @param {Boolean} enabled
+     * @memberOf AddonsItem
+     */
+    set enabled(enabled) {
+      if (this._enabled === enabled) {
+        return;
+      } else {
+        this._enabled = enabled;
+        if (this._enabled) {
+          this._updateAddonSectionVisibility();
+        }
+      }
+    },
+
+    /**
+     * Get current status of addonsItem
+     *
+     * @access public
+     * @memberOf AddonsItem
+     */
+    get enabled() {
+      return this._enabled;
+    },
+
+    /**
+     * Initialization
+     *
+     * @access private
+     * @memberOf AddonsItem
+     * @return {Promise}
+     */
+    init: function() {
+      this.addonsManager.init().then( () => {
+        this.addonsManager.addons.observe('apps',
```

Comment:
This won't work as `addonsManager.addons` is not an `ObservableArray`. However, currently `ObservableArray` does not have an event notifying the change of length, so we may need to keep the count of the addons in `AddonsManager`.